stdlib/list.ml

```diff
@@ -77,6 +77,15 @@ let rec flatten = function
 
 let concat = flatten
 
+let[@tail_mod_cons] rec intersperse sep = function
+  | ([] | [ _ ]) as l -> l
+  | a :: (_ :: _ as l) -> a :: sep :: intersperse sep l
+
+let[@tail_mod_cons] rec intercalate sep = function
+  | [] -> []
+  | [a] -> a
+  | a1::a2::l -> a1 @ sep @ intercalate sep (a2 :: l)
```

Comment:
This function is not TRMC as the recursive call is not argument to a constructor.

Reply:
Right. I've re-run all the tests locally but forgotten to rebuild everything at the root level, so there wasn't any issue on my computer.

Also, unrelated to the PR: how do I rebuild changes to the standard library during experimentation without cleaning (`make clean && make distclean` at the root directory) and rebuilding everything from scratch? The process seems to be taking quite a lot of time. If I avoid cleaning, I get weird errors, something like "inconsistent assumptions about the module X".